addons/mass_mailing/wizard/mail_compose_message.py

```diff
@@ -40,11 +40,29 @@ def get_mail_values(self, res_ids):
                         'mailing_model': self.model,
                         'mailing_domain': self.active_domain,
                 })
+
+            # Preprocess res.partners to batch-fetch from db
+            # if recipient_ids is present, it means they are partners 
+            # (the only object to fill get_default_recipient this way)
+            recipient_partners_ids = []
+            read_partners = {}
+            for res_id in res_ids:
+                mail_values = res[res_id]
+                if mail_values.get('recipient_ids'):
+                    # recipient_ids is a list of tuples at this point
+                    recipient_partners_ids.append(mail_values.get('recipient_ids')[0][1])
```

Comment:
I'm not very fond of that ugly command syntax, but I guess you kept this approach (rather than switching to a pure list of IDs) to avoid side-effects? Well, that will do for now. The whole thing is due for a refactoring anyway.